src/plugins/data/public/search/aggs/agg_config.ts

```diff
@@ -27,18 +28,20 @@ import { ISearchSource } from '../search_source';
 import { FieldFormatsContentType, KBN_FIELD_TYPES } from '../../../common';
 import { FieldFormatsStart } from '../../field_formats';
 
-export interface AggConfigOptions {
-  type: IAggType;
+export interface AggConfigJson {
```

Comment:
why is it named AggConfigJSON ? its not json :)

Reply:
it was because it is the shape that's returned from `toJSON()` (which as you correctly point out below, is confusing in the first place.

my vote would be to call the method `serialize()` and the interface `AggConfigSerialized`. we can keep `toJSON()` temporarily for BWC, though it doesn't seem to be used in a bunch of places anyway.